src/Symfony/Component/DependencyInjection/ParameterBag/EnvPlaceholderParameterBag.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\ParameterBag;
+
+use Symfony\Component\DependencyInjection\Exception\RuntimeException;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class EnvPlaceholderParameterBag extends ParameterBag
+{
+    private $envPlaceholders = array();
+
+    /**
+     * {@inheritdoc}
+     */
+    public function get($name)
+    {
+        if (0 === strpos($name, 'env(') && ')' === substr($name, -1) && 'env()' !== $name) {
+            $env = substr($name, 4, -1);
+
+            if (isset($this->envPlaceholders[$env])) {
+                return $this->envPlaceholders[$env];
+            }
+            if (!preg_match('/^[A-Z_]+$/i', $env)) {
+                throw new InvalidArgumentException(sprintf('Invalid %s name: only characters "A-Z" and "_" are allowed.', $name));
+            }
+
+            if ($this->has($name)) {
+                $defaultValue = parent::get($name);
+
+                if (!is_scalar($defaultValue)) {
+                    throw new RuntimeException(sprintf('The default value of an env() parameter must be string, but "%s" given to "%s".', gettype($defaultValue), $name));
```

Comment:
Something that can be concatenated is fine, but the error message is easier to act on this way to me.